banyan/src/forest/read.rs

```diff
@@ -110,50 +150,31 @@ where
 
     fn next(&mut self) -> Option<Self::Item> {
         let res: FilteredChunk<T, V, E> = 'outer: loop {
-            let head = match self.index_stack.last() {
+            let head = match self.stack.last_mut() {
```

Comment:
`let head = if let Some(i) = self.stack.last_mut() { i } else { return None }` ? Not sure if it is better...